### PR TITLE
OD-1850 Taxjar communication error in the case of US invoice in EURO

### PR DESCRIPTION
**Pull request by astirpe**
This is the fix for the wrong discount computation due to the currency conversion, as described in https://vmbikes.atlassian.net/browse/OD-1850

I added an automated test to reproduce the currency conversion problem.
In that test you can notice that the correct final discount amount is expected to be 49.00 USD.
Without this proposed fix, the computed discount amount is 48.95 USD which is wrong.

### CODE REVIEW

File: taxjar_salestax/models/taxjar_config.py [hunk 1]
```diff
@@ -212,6 +216,17 @@ def create_order(
             _logger.debug('Response: %s', res)
         return res
 
+    @api.model
+    def _get_discount_percentage_from_order_item(self, item):
```
Comment (dkubiak789) [inline taxjar_salestax/models/taxjar_config.py:220]: I believe it should be `_get_discount_amount_from_order_item` and `discount_amount`.

**Review by dkubiak789 — APPROVED** [review]

File: taxjar_salestax/models/taxjar_config.py [hunk 2]
```diff
@@ -183,10 +183,14 @@ def create_order(
                 'An order without any lines cannot be created in TaxJar')
 
         for item in items:
-            for key in ['unit_price', 'discount', 'sales_tax']:
+            discount_perc = self._get_discount_percentage_from_order_item(item)
```
Comment (StefanRijnhart) [inline taxjar_salestax/models/taxjar_config.py]: I think you can use `discount_perc = self.env['account.invoice.line'].browse(item['id']).discount` here
(`create_order` is taxjar terminology, and is only applied to invoices, not to sale orders)

Comment (StefanRijnhart) [reply taxjar_salestax/models/taxjar_config.py]: Or you can add the discount percentage in https://github.com/VanMoof/account-fiscal-rule/blob/add/12.0/taxjar_salestax/taxjar_salestax/models/account_invoice_line.py#L42-L49 and `pop` it from the item here.

Comment (astirpe) [reply taxjar_salestax/models/taxjar_config.py]: :+1: thanks!

**Review by StefanRijnhart — APPROVED** [review]

**Pull request merged**